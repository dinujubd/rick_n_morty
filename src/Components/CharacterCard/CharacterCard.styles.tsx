import { createStyles, makeStyles, Theme } from "@material-ui/core";

export const useStyles = (isMobile: boolean) => makeStyles((theme: Theme) =>
    createStyles({
        root: {
            borderRadius: '0',
            boxShadow: 'none'
        },
        container: {
            height: isMobile ? 'auto': '90vh'
        },
        cover: {
            objectFit: 'cover',
            maxHeight: isMobile ? 'auto': '45%'
        },
        name: {
            fontWeight: 900
        },
        dataTitle: {
            fontWeight: 'bold'
        },
        nameBlock: {
            display: 'flex',
        },
        genderIcon: {
            marginLeft: 5,
            height: 20
        },
        details: {
            padding: theme.spacing(2)
        },
        lifeStatus: {
            height:
                40
        }
    }),
);