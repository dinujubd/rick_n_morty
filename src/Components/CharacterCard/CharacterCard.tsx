import * as React from 'react';
import { Box, Card, CardContent, CardMedia, createStyles, Grid, makeStyles, Theme, Typography } from '@material-ui/core';
import { connect } from 'react-redux';
import { Character } from '../../models/characters';
import { ApplicationState } from '../../store/store';
import Male from '../../assets/male.png'
import Female from '../../assets/female.png'
import Dead from '../../assets/dead.png'
import Alive from '../../assets/alive.png'
import Unknown from '../../assets/unknown.png'

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            borderRadius: '0',
            boxShadow: 'none'
        },
        cover: {
            objectFit: 'cover',
            height: '65%'
        },
        name: {
            fontWeight: 900
        },
        dataTitle: {
            fontWeight: 'bold'
        },
        nameBlock: {
            display: 'flex',
            borderRadius: '50px'
        },
        genderIcon: {
            marginLeft: 5
        },
        details: {
            padding: '1em'
        },
    }),
);


interface StateProps {
    character?: Character,
}

const CharacterCard: React.FC<StateProps> = ({ character }) => {
    const classes = useStyles();

    if (!character) return null;

    const getGenderImage = () => {
        switch (character.gender) {
            case 'Male': return Male;
            case 'Female': return Female;
            case 'Unknown': return Unknown;
        }
    }

    const getLifeStatus = () => {
        switch (character.status) {
            case 'Dead': return Dead;
            case 'Alive': return Alive;
        }
    }

    return (
        <Card className={classes.root}>

            <Grid container
                direction="column"
                style={{height: '90vh'}}
            >
                <img className={classes.cover} alt={character.name} src={character.image} />

                <Grid
                    container
                    justify="space-between"
                    className={classes.details}
                >
                    <Box className={classes.nameBlock}>
                        <Typography className={classes.name} variant="h5" component="h2">
                            {character.name}
                        </Typography>

                        <img className={classes.genderIcon} style={{ height: 20 }} src={getGenderImage()} />
                    </Box>
                    <img style={{ height: 40 }} src={getLifeStatus()} />

                </Grid>

                <Grid
                    container
                    alignItems="stretch"
                    className={classes.details}
                >
                    <Grid
                        container
                        justify="space-between"
                    >
                        <Typography className={classes.dataTitle} variant="body1" gutterBottom>
                            Species
                        </Typography>
                        <Typography variant="body1" gutterBottom>
                            {character.species}
                        </Typography>

                    </Grid>

                    <Grid
                        container
                        justify="space-between"
                    >
                        <Typography className={classes.dataTitle} variant="body1" gutterBottom>
                            Location
                        </Typography>
                        <Typography variant="body1" gutterBottom>
                            {character.location.name}
                        </Typography>

                    </Grid>

                    <Grid
                        container
                        justify="space-between"
                    >
                        <Typography className={classes.dataTitle} variant="body1" gutterBottom>
                            Origin
                        </Typography>
                        <Typography variant="body1" gutterBottom>
                            {character.origin.name}
                        </Typography>

                    </Grid>

                </Grid>
                 
            </Grid>
        </Card>
    )
}

CharacterCard.displayName = 'CharacterCard';

const mapStateToProps = (state: ApplicationState): StateProps => ({
    character: state.characters.currentCharacter
})

export default connect(mapStateToProps)(CharacterCard);