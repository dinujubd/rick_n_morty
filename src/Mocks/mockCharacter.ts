import { Character } from "../Models/characters";

const mockCharacter: Character = {
    id: 1,
    name: "Rick Sanchez",
    status: "Alive",
    species: "Human",
    type: "",
    gender: "Male",
    origin: {
        name: "Earth (C-137)",
        url: "https://rickandmortyapi.com/api/location/1"
    },
    location: {
        id: 1,
        dimension: '',
        residents: [],
        name: "Earth (Replacement Dimension)",
        url: "https://rickandmortyapi.com/api/location/20"
    },
    image: "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
    episode: [],
    url: "https://rickandmortyapi.com/api/character/1",
    created: new Date("2017-11-04T18:48:46.250Z")
}

export default mockCharacter;


