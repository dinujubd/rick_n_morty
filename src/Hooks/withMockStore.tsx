import * as React from 'react';

import configureMockStore from "redux-mock-store";
import { Provider } from 'react-redux';
import mockCharacter from '../Mocks/mockCharacter';


export const withMockStore = (component: React.ReactElement) => {

    const mockStore = configureMockStore();
    const store = mockStore({characters: {
        characters: [{mockCharacter}],
        currentCharacter: mockCharacter
    }});

    return (
        <Provider store={store}>
            {component}
        </Provider>
    );

}