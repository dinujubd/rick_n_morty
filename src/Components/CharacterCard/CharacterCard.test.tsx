import renderer from 'react-test-renderer';
import { CharacterCard } from './CharacterCard';
import mockCharacter from '../../Mocks/mockCharacter';
import { withMockStore } from '../../Hooks/withMockStore';

describe('<CharacterCard />', () => {
    it('should renders correctly', () => {
        const tree = renderer
            .create(withMockStore(<CharacterCard character={mockCharacter}/>))
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});