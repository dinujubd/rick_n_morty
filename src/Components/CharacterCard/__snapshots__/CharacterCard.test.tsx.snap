// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharacterCard /> should renders correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-10 makeStyles-root-1"
>
  <div
    className="MuiBox-root MuiBox-root-11"
  >
    <div
      className="MuiBox-root MuiBox-root-12"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-8"
      >
        <img
          className="MuiAvatar-img"
          src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
        />
      </div>
    </div>
  </div>
  <div
    className="MuiBox-root MuiBox-root-13"
  >
    <div
      className="MuiGrid-root makeStyles-details-7 MuiGrid-container MuiGrid-justify-xs-space-between"
    >
      <h2
        className="MuiTypography-root makeStyles-name-4 MuiTypography-h5"
      >
        RICK SANCHEZ
      </h2>
      <img
        alt="Rick Sanchez"
        className="makeStyles-genderIcon-6"
        src="male.png"
      />
    </div>
    <div
      className="MuiGrid-root makeStyles-details-7 MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-between"
      >
        <div
          className="MuiBox-root MuiBox-root-14 makeStyles-dataTitle-5"
        >
          <span
            className="MuiBox-root MuiBox-root-15"
          >
            Featured Episodes:
          </span>
          <span
            className="MuiBox-root MuiBox-root-16 makeStyles-castedCount-9"
          >
            [0]
          </span>
        </div>
        <span
          className="MuiBox-root MuiBox-root-18"
        >
          <span
            className="MuiBox-root MuiBox-root-19"
          >
            Alive
          </span>
          <span
            className="MuiBox-root MuiBox-root-20 makeStyles-root-17"
          />
        </span>
      </div>
      <div
        className="MuiBox-root MuiBox-root-22"
      >
        <div
          className="MuiBox-root MuiBox-root-23 makeStyles-dataTitle-21"
        >
          Location
        </div>
        <div
          className="MuiBox-root MuiBox-root-24"
        >
          <span>
            Earth (Replacement Dimension)
            , Dimension: 
            , Residents: 
          </span>
        </div>
      </div>
      <div
        className="MuiBox-root MuiBox-root-25"
      >
        <div
          className="MuiBox-root MuiBox-root-26 makeStyles-dataTitle-21"
        >
          Origin
        </div>
        <div
          className="MuiBox-root MuiBox-root-27"
        >
          <span>
            Earth (C-137)
            , Dimension: 
            , Residents: 
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
